src/meshlink.c

```diff
@@ -1314,6 +1161,17 @@ meshlink_handle_t *meshlink_open_ex(const meshlink_open_params_t *params) {
 		mesh->name = xstrdup(params->name);
 	}
 
+	// Hash the key
+	if(params->key) {
+		mesh->config_key = xmalloc(CHACHA_POLY1305_KEYLEN);
+
+		if(!prf(params->key, params->keylen, "MeshLink configuration key", 26, mesh->config_key, CHACHA_POLY1305_KEYLEN)) {
+			logger(NULL, MESHLINK_ERROR, "Error creating configuration key!\n");
+			meshlink_errno = MESHLINK_EINTERNAL;
```

Comment:
In the event of failure, shouldn't we be freeing the `mesh->config_key` and setting it to NULL before returning.